### PR TITLE
Completed array-manipulation-3.

### PR DESCRIPTION
**Pull request by alexdenbaugh**
array-manipulation-3

### CODE REVIEW

File: array-manipulation-3/src/union.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/* exported union */
+function union(first, second) {
+  var newArray = [];
+  for (var i = 0; i < first.length; i++) {
+    newArray.push(first[i]);
+  }
```
Comment (codingcodymiller) [inline array-manipulation-3/src/union.js:6]: Same feedback here! What if `first` includes multiple values that are the same? We only want one of each value in the new array to be returned.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]
Nice work so far!! Please see my notes

File: array-manipulation-3/src/flatten.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  }
+  for (var i = 0; i < array.length; i++) {
+    if (Array.isArray(array[i])) {
+      if (array[i].length > 1) {
+        for (j = 0; j < array[i].length; j++) {
+          newArray.push(array[i][j]);
+        }
+      } else {
+        newArray.push(array[i][0]);
+      }
```
Comment (scbowler) [inline array-manipulation-3/src/flatten.js:16]: The extra check of checking the array length isn't necessary and has no net benefit.

File: array-manipulation-3/src/zip.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/* exported zip */
+function zip(first, second) {
+  var newArray = [];
+  var pairArray = [];
+  for (var i = 0; first[i] && second[i]; i++) {
+    pairArray = [];
+    pairArray.push(first[i]);
+    pairArray.push(second[i]);
```
Comment (scbowler) [inline array-manipulation-3/src/zip.js:8]: You can pass multiple things to `.push`

For Example:
```JAVASCRIPT
pairArray.push('Item 1', 'Item 2', 'Item 3');
```

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking really good, a couple cleanup items needed. Let us know if you have any questions.

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍

**Pull request merged**